GiEnJul/ClientApp/src/components/institution/FormPerson.tsx

```diff
@@ -16,132 +16,164 @@ import { isNotNull } from "../InputFields/Validators/Validators";
 import IFormPerson from "./IFormPerson";
 
 interface PersonProps {
-	updatePerson: (newPerson: IFormPerson) => void;
-	deletePerson: () => void;
-	viewErrorTrigger: number;
-	person: IFormPerson;
+  updatePerson: (newPerson: IFormPerson) => void;
+  deletePerson: () => void;
+  viewErrorTrigger: number;
+  person: IFormPerson;
 }
 
+const initFormDataState: { [data: string]: any } = {
+  age: "",
+  gender: Gender.Unspecified,
+  wish: "",
+  ageWish: false,
+};
+
+const initValidFormState: { [valid: string]: boolean } = {
+  age: false,
+  gender: false,
+  wishInput: false,
+  wish: false,
+};
+
 const InstitutionPerson: FC<PersonProps> = ({
-	updatePerson,
-	deletePerson,
-	viewErrorTrigger,
-	person,
+  updatePerson,
+  deletePerson,
+  viewErrorTrigger,
+  person,
 }) => {
-	const [age, setAge] = useState("");
-	const [gender, setGender] = useState(Gender.Unspecified);
-	const [wish, setWish] = useState("");
-	const [isAgeWish, setIsAgeWish] = useState(false);
-	const [isValidAge, setIsValidAge] = useState(false);
-	const [isValidGender, setIsValidGender] = useState(false);
-	const [isValidWishInput, setIsValidWishInput] = useState(false);
-	const [isValidWish, setIsValidWish] = useState(false);
+  const [formDataState, setFormDataState] = useState(initFormDataState);
+  const [validFormState, setValidFormState] = useState(initValidFormState);
+
+  const getFormDataSetter = (target: keyof typeof formDataState) => {
+    return (value: unknown) => {
+      setFormDataState((prev) => {
+        {
+          prev[target] = value;
+          return { ...prev };
+        }
```

Comment:
```suggestion
          prev[target] = value;
          return { ...prev };
```
Could this bracket pair be removed?

Reply:
Yes it could! I see that this applies to other places as well.